### PR TITLE
Addon-docs: Doc blocks fixes for dark theme

### PR DESCRIPTION
**Pull request by atanasster**
Issue: #7982

## What I did
set color in DocPage to `theme.color.defaultText,` to accommodate for dark theme

## How to test
set `options.theme: themes.dark,` (wait if local storage still not refreshed)

### CODE REVIEW

**Comment by atanasster** [comment]
screenshot after change:

screenshot820

**Review by shilman — CHANGES_REQUESTED** [review]
I don't think this is the right solution unfortunately.

We have a complicated situation where we don't want the global styles to interfere with the user's story styles.

As far as I understand, this means that every doc block should be independently themable. So AFAIK the solution is to fix them theming for `Description` and `PropsTable` separately, rather than fix it at the top level as you've don here (although that's simpler).

@domyen can you chime in here?

**Comment by atanasster** [comment]
@shilman Initially I did change the PropsTable and Markdown elements, but then noticed that `DocsWrapper` already has css styling (thats how the background is set to `theme.background.content`), so the color prop does not change the behavior:
```
export const DocsWrapper = styled.div<{}>(({ theme }) => ({
  background: theme.background.content,
  color: theme.color.defaultText,
...
```

**Comment by ndelangen** [comment]
This color property will cascade into user's components, where they might expect it to be the browser default.

I think there should be a place where we set the font-family. Color belongs there too probably.

**Comment by atanasster** [comment]
@ndelangen - `DocsWrapper` was already setting the background color and it was propagating to the user components and other elements of the docs page, causing actually the initial issue with text not visible on dark theme. This PR is only adding the text color prop to match the background.

ISTM that either background should be removed from here or color is rightly placed also here.

**Comment by shilman** [comment]
@atanasster Does the background also propagate like that? Can you double check? I don't think it does.

**Comment by domyen** [comment]
As Michael said we have a complex situation (due to inline rendering + style bleeding). That means we'll need to theme doc blocks independently. 

For DocsWrapper the background will not cascade into child elements but `color` will. I'd suggest adding `color` to each doc block you want to theme.

**Comment by atanasster** [comment]
@domyen - thanks for the input, switched to setting colors for the individual elements (hope I didn't miss too many :)).

It seems that there is an issue with user text in stories once we are not setting it anymore, ie:
```
storiesOf('Basics|Button', module).add('all buttons', () => (
  <div>
    <p>Button that is used for forms</p>
    <FormButton>Form.Button</FormButton>
    <br />
```
will be like that: 
screenshot821

the same on the preview window:
screenshot822

**Comment by domyen** [comment]
Thanks Atanas, unfortunately the first screenshot is an unavoidable situation because we shouldn't actually style any of the users code ourselves. When using the dark theme users need to write their stories to accommodate dark backgrounds.

In the case of SB design system, we wrote our stories for the light theme and supply a background when possible, that's why they look that way. 

It's not perfect as you pointed out, but that's probably something we need to fix in the design system code.

**Comment by atanasster** [comment]
So what about the background from `DocWrapper` ?

The background color is set in multiple places (but not the text color), here is the DOM:
screenshot823

IframeWrapper - outside the iFrame:
```
const IframeWrapper = styled.div(({ theme }) => ({
  position: 'absolute',
  top: 0,
  left: 0,
  bottom: 0,
  right: 0,
  width: '100%',
  height: '100%',
  background: theme.background.content,
}));
```
DocsWrapper - inside the iFrame, also changes the color (but if its only removed then the color stays because of higher-up in hierarchy):
```
export const DocsWrapper = styled.div<{}>(({ theme }) => ({
  background: theme.background.content,
  display: 'flex',
  justifyContent: 'center',
  minHeight: '100vh',
  padding: '4rem 20px',

  [`@media (min-width: ${breakpoint * 1}px)`]: {},
}));
```

File: lib/components/src/typography/DocumentFormatting.tsx [hunk 1]
```diff
@@ -25,24 +25,29 @@ const withMargin: CSSObject = {
 export const H1 = styled.h1<{}>(withReset, headerCommon, ({ theme }) => ({
   fontSize: `${theme.typography.size.l1}px`,
   fontWeight: theme.typography.weight.black,
+  color: theme.color.defaultText,
```
Comment (shilman) [inline lib/components/src/typography/DocumentFormatting.tsx]: This can be put in headerCommon

Comment (atanasster) [reply lib/components/src/typography/DocumentFormatting.tsx]: thanks, great feedback. Updated `headerCommon` and pushed

**Review by shilman — APPROVED** [review]
LGTM!

**Pull request merged**